app/item/[id]/page.tsx

```diff
@@ -61,8 +61,7 @@ export default async function listing({
       .from('users')
       .select('favourite_items')
       .eq('id', userID)
-    const favouriteItems: number[] | null =
-      favourites && favourites[0].favourite_items
+    const favouriteItems = favourites && favourites[0].favourite_items
```

Comment:
I haven't systematically been through but I think there a few places like this where you shouldn't need to declare the types so you can just get rid of them